release/models/interfaces/openconfig-if-ip.yang

```diff
@@ -514,6 +520,17 @@ module openconfig-if-ip {
       description
         "The length of the subnet prefix.";
     }
+
+    leaf type {
+      type oc-inet:ipv6-address-type;
+      default GLOBAL_UNICAST;
+      description
+        "Specifies the explicit type of the IPv6 address being assigned
+        to the subinterface. By default, addresses are assumed to be
+        global unicast.  Where a link-local address is to be explicitly
+        configured, this leaf should be set to LINK_LOCAL.";
```

Comment:
I belive we shall explicitly describe behaviour, if addres configured is not mathcing address type. That is What is expected behaviour if:
A) address is NOT FE80::/10 AND ipv6-address-type is LINK_LOCAL_UNICAST
B) address is form FE80::/10 AND ipv6-address-type is GLOBAL_UNICAST (or omited)

Reply:
If the target is compliant with my interpretation of [RFC4291](https://www.rfc-editor.org/rfc/rfc4291.html#section-2.5.6), I expect A and B would both be rejected.

Reply:
@rszarecki I updated the description for the new types to state it is required that the value for the  address must match the type as defined in RFC4291.